cli/command/service/scale.go

```diff
@@ -10,19 +10,28 @@ import (
 	"github.com/docker/cli/cli"
 	"github.com/docker/cli/cli/command"
 	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/versions"
 	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
 func newScaleCommand(dockerCli *command.DockerCli) *cobra.Command {
-	return &cobra.Command{
+	options := newServiceOptions()
+
+	cmd := &cobra.Command{
 		Use:   "scale SERVICE=REPLICAS [SERVICE=REPLICAS...]",
 		Short: "Scale one or multiple replicated services",
 		Args:  scaleArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runScale(dockerCli, args)
+			return runScale(dockerCli, cmd.Flags(), options, args)
 		},
 	}
+
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.detach, "detach", "d", true, "Exit immediately instead of waiting for the service to converge")
+	flags.SetAnnotation("detach", "version", []string{"1.29"})
```

Comment:
We could move these two lines to an `addDetachFlag()` function and call it from `addServiceFlags()` (and here).

We should also create a constant `flagDetach` like we have for a bunch of  the other flags in `service/opts.go`